python/core/qgsvectordataprovider.sip

```diff
@@ -401,7 +401,22 @@ class QgsVectorDataProvider : QgsDataProvider
      * Get some metadata that will be display in the metadata tab of the layer properties.
      * @return The provider metadata
      */
-    virtual QMap<QString, QString> metadata() const;
+    virtual QVariantMap metadata() const;
+
+    /**
+     * Get the translated metadata key.
+     * @param md_key The metadata key
+     * @return The translated metadata value
+     */
+    virtual QString translatedMetadataKey( QString md_key ) const;
```

Comment:
* Imperatively call it translateMetadataKey() and translateMetadataValue() (they are more transformers than getters)
* pass a `const QString& mdKey ` as parameter